src/bp/common/http.ts

```diff
@@ -13,6 +13,21 @@ const escapeHtmlSimple = (str: string) => {
     .replace(/`/g, '&#96;')
 }
 
+export const URL_FOLDER_SEPERATOR = '!!'
+
+const escapeRegExp = (str: string) => str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
+
+const decoded_folder_seperator_re = /\//g
+const encoded_folder_seperator_re = new RegExp(escapeRegExp(URL_FOLDER_SEPERATOR), 'g')
+
+export const encodeFolderPath = (path: string): string => {
+  return path.replace(decoded_folder_seperator_re, URL_FOLDER_SEPERATOR)
+}
+
+export const decodeFolderPath = (urlPath: string): string => {
+  return urlPath.replace(encoded_folder_seperator_re, '/')
+}
```

Comment:
Is the escapeRegExp really necessary? Also, you can put the reg exp directly where you use them, it doesn't add value to put them in a separacte const, just makes it harder to understand.

 I think you could replace these lines with this:
```js
export const encodeFolderPath = (path: string): string => {
  return path.replace(/\//g, URL_FOLDER_SEPERATOR)
}
export const decodeFolderPath = (urlPath: string): string => {
  return urlPath.replace(new RegExp(URL_FOLDER_SEPERATOR, 'g'), '/')
}
```

Reply:
Fixed, these made more sense back when I chose a more complicated folder separator